### PR TITLE
Fix ingestion (col values gets overwritten due to parsing buffer)

### PR DESCRIPTION
**Pull request by vasukalariya**
# Description
Summarize the change.
- Do not use parsing buffer if there is an escape character.

Fixes #<issue-number> (link all the GitHub issues this addresses)

# Testing
Describe how you tested this code. How can the reviewers reproduce your tests?

# Checklist:
Before marking your pull request as ready for review, complete the following.

- [ ] I have self-reviewed this PR.
- [ ] I have removed all print-debugging and commented-out code that should not be merged.
- [ ] I have added sufficient comments in my code, particularly in hard-to-understand areas.
- [ ] I have formatted the code, if applicable. For Go, I have run `goimports -w .`.

### CODE REVIEW

File: pkg/segment/writer/logpacker.go [hunk 1]
```diff
@@ -167,7 +166,8 @@ func parseSingleString(key string, tsKey *string, valBytes []byte, ple *ParsedLo
 	cbufidx += 1
 	n := uint16(len(valBytes))
 	utils.Uint16ToBytesLittleEndianInplace(n, ple.allCvalsTypeLen[ple.numCols][cbufidx:])
-	ple.allCvals[ple.numCols] = valBytes
+	ple.allCvals[ple.numCols] = make([]byte, n)
+	copy(ple.allCvals[ple.numCols], valBytes)
```
[the remark this answers is not in the document]
Comment (AndrewHess) [reply pkg/segment/writer/logpacker.go:170]: I agree. Is it relatively easy to add a unit test or some other automated test for this that fails without this PR but passes with this PR?

**Review by AndrewHess — APPROVED** [review]

File: pkg/segment/writer/logpacker.go [hunk 2]
```diff
@@ -113,7 +119,6 @@ func parseNonJaegerRawJsonArray(currKey string, data []byte, tsKey *string,
 				return
 			}
 		case jp.String:
-
 			valUnescaped, encErr := jp.Unescape(value, jsParsingStackbuf[:])
```
Comment (AndrewHess) [inline pkg/segment/writer/logpacker.go:122]: Do you need a similar fix here like you added in `parseRawJsonObject()`?

File: pkg/segment/writer/logpacker.go [hunk 3]
```diff
@@ -51,12 +52,17 @@ func ParseRawJsonObject(currKey string, data []byte, tsKey *string,
 				return fmt.Errorf("parseRawJsonObject: arr currKey: %v, err: %v", currKey, err)
 			}
 		case jp.String:
-			valUnescaped, err := jp.Unescape(value, jsParsingStackbuf[:])
-			if err != nil {
-				return fmt.Errorf("parseRawJsonObject: failed to unescape currKey: %v, err: %v",
-					currKey, err)
+			firstBackslash := bytes.IndexByte(value, '\\')
+			if firstBackslash != -1 {
+				valUnescaped, err := jp.Unescape(value, nil)
```
Comment (AndrewHess) [inline pkg/segment/writer/logpacker.go:57]: I'm not seeing how the new code will ever give a different result than the old code, and I'm not really sure why we're doing this. Maybe add a comment about the purpose of this code (i.e., what issue you're fixing). Also, are you able to add some automated test so we don't regress on this issue? It seems like a pretty subtle bug

File: pkg/segment/writer/logpacker.go (hunk 2 above)
Comment (vasukalariya) [reply pkg/segment/writer/logpacker.go]: addressed in the latest commit.

**Review by nkunal — APPROVED** [review]

**Pull request merged**